### PR TITLE
Add mesmer recipe

### PR DESCRIPTION
**Pull request by znicholls**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by znicholls** [comment]
@mathause are you happy to be a maintainer?

File: recipes/mesmer/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    - pip check
+    - python -c "import mesmer; assert mesmer.__version__ == '{{ version }}', mesmer.__version__"
+
+about:
+  home: https://github.com/openscm/mesmer
+  license: GPL-3.0-or-later
+  license_family: GPL
+  license_file: LICENSE
+  summary: Modular Earth System Model Emulator with spatially Resolved output
+  doc_url: https://mesmer.readthedocs.io/en/stable/
```
Comment (mathause) [inline recipes/mesmer/meta.yaml:49]:
```suggestion
  doc_url: https://mesmer-emulator.readthedocs.io/en/stable/
```

File: recipes/mesmer/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  dev_url: https://github.com/openscm/mesmer
```
Comment (mathause) [inline recipes/mesmer/meta.yaml:50]:
```suggestion
  dev_url: https://github.com/MESMER-group/mesmer
```

**Review by mathause — COMMENTED** [review]
Yes, sure!

File: recipes/mesmer/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  imports:
+    - mesmer
+  requires:
+    - pip
+  commands:
+    - pip check
+    - python -c "import mesmer; assert mesmer.__version__ == '{{ version }}', mesmer.__version__"
+
+about:
+  home: https://github.com/openscm/mesmer
```
Comment (dopplershift) [inline recipes/mesmer/meta.yaml:44]: Should this be:
```suggestion
  home: https://github.com/MESMER-group/mesmer
```
? The other one 404's.

File: recipes/mesmer/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+build:
+  number: 0
+  noarch: python
+  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed -vvv"
+
+requirements:
+  host:
+    - pip
+    - python >=3.6
```
Comment (dopplershift) [inline recipes/mesmer/meta.yaml:21]: Upstream only lists >=3.7, not sure if that's important or not.

Comment (znicholls) [reply recipes/mesmer/meta.yaml:21]: oops good pick up 

```suggestion
    - python >=3.7
```

File: recipes/mesmer/meta.yaml (hunk 3 above)
Comment (znicholls) [inline recipes/mesmer/meta.yaml]:
```suggestion
  home: https://github.com/MESMER-group/mesmer
```

**Comment by znicholls** [comment]
Thanks for the review @dopplershift! I think this should be fixed now

File: recipes/mesmer/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  number: 0
+  noarch: python
+  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed -vvv"
+
+requirements:
+  host:
+    - pip
+    - python >=3.7
+  run:
+    - python >=3.6
```
Comment (dopplershift) [inline recipes/mesmer/meta.yaml:23]: Missed this one
```suggestion
    - python >=3.7
```

**Review by dopplershift — CHANGES_REQUESTED** [review]

File: recipes/mesmer/meta.yaml (hunk 5 above)
Comment (znicholls) [reply recipes/mesmer/meta.yaml]: so close, yet so far

**Review by dopplershift — APPROVED** [review]

**Pull request merged**